### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by normanjoyner**
drone-containership plugin implementation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.PHONY: clean deps test build docker
```
Comment (tboerger) [inline Makefile]: Please use the updated Makefile used by most plugins

File: main.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	"fmt"
+	"os"
+
+	"github.com/containership/go-containership.cloud/client"
+	"github.com/drone/drone-go/drone"
+	"github.com/drone/drone-go/plugin"
+)
+
+var (
+	build     string
```
Comment (tboerger) [inline main.go]: This variable have been dropped within the other plugins

File: main.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		fmt.Println("Error: ContainerShip Cloud API Key is required!")
+		os.Exit(1)
+	}
+
+	if vargs.ClusterId == "" {
+		fmt.Println("Error: ContainerShip Cloud Cluster ID is required!")
+		os.Exit(1)
+	}
+
+	if vargs.Image == "" {
```
Comment (tboerger) [inline main.go:40]: I would default to `repo.FullName+":latest"`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,101 @@
...
+## Binary
+
+Build the binary using `make`:
+
+```
+make deps build
+```
+
+### Example
+```sh
```
Comment (tboerger) [inline README.md:19]: NITPICK: Please add an empty line :)

File: README.md [hunk 5]
```diff
@@ -1,2 +1,101 @@
 # drone-containership
 Drone plugin for deploying to ContainerShip
```
Comment (tboerger) [inline README.md:7]: Please add the badges that are part of the other plugins as well

**Comment by normanjoyner** [comment]
@tboerger I think all the requested changes have been made. Let me know if there is anything else I can do!

File: main.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		vargs.Image = repo.FullName + ":latest"
```
Comment (tboerger) [inline main.go:41]: Just some personal taste but I prefer to move the default assignments to the top behind `plugin.MustParse()`

File: main.go [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	if vargs.Organization == "" {
+		fmt.Println("Error: ContainerShip Cloud Organization is required!")
+		os.Exit(1)
+	}
+
+	ContainerShipCloudClient := client.NewContainerShipCloudClient(vargs.Organization, vargs.ApiKey)
```
Comment (tboerger) [inline main.go:49]: NITPICK: I would just name it `client` instead of `ContainerShipCloudClient`

File: main.go [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	application := map[string]interface{}{
+		"id":    vargs.Application,
+		"image": vargs.Image,
+	}
+
+	if response.StatusCode == 404 {
```
Comment (tboerger) [inline main.go:63]: NITPICK: Maybe simply use a switch, e.g.

``` go
switch code := response.StatusCode {
case 404:

case 200:

default:

}
```

**Comment by tboerger** [comment]
Some tiny nitpicking, otherwise I like it. Thanks for the work!
LGTM

**Comment by tboerger** [comment]
I will merge this PR for now, feel free to fix the nitpicking in a followup PR.

**Comment by normanjoyner** [comment]
thanks! I will get these fixed up today

**Pull request merged**